AdblockPlusSafariExtension/AdblockPlus+Parsing.m

```diff
@@ -0,0 +1,308 @@
+/*
+ * This file is part of Adblock Plus <https://adblockplus.org/>,
+ * Copyright (C) 2006-2015 Eyeo GmbH
+ *
+ * Adblock Plus is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * Adblock Plus is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Adblock Plus.  If not, see <http://www.gnu.org/licenses/&gt.
+ */
+
+#import "AdblockPlus+Parsing.h"
+
+// yajl is sax-like json parser. Content blocker extension has limited amount of memory,
+// so that it is not possible to load whole filter list at once.
+#include <yajl/yajl_parse.h>
```

Comment:
The main reason is that the current solution is more robust. I spent lot of time thinking about that. There are two main reasons:

1) The first problem are embed files with filter lists. They might be stored in either content blocker or host application but not in both (because otherwise they would have to be stored two times and those files are quite big). One app would have to copy those files into shared folder after startup.

2) Now filter list files are created on demand. They do not have to processed in advance. If the host app has crashed or been killed, host app does not have to be run again to recompute files.

I can move merge logic into host application. But it would mean that I will have to handle lot of edge cases, where the host application was started before/after content blocker and situations, when it does not finished processing.